### DISCUSSION TITLE
Aeotec Range Extender 7 (ZW189-A01) - Night light function?

### DISCUSSION THREAD

**Discussion by Anto79-ops** [zwave-js/zwave-js#4954] [Ideas]
Hello,

Got a few of the Aeotec Range Extender 7 (ZW189-A01). Paired and works.

https://devices.zwave-js.io/?jumpTo=0x0371:0x0004:0x00bd:0.0

The soft action button is also a white light:

![image](https://user-images.githubusercontent.com/79593761/187034954-21c9132d-2d6c-44f2-a838-257ea60bccb3.png)

The light can be turned off and on by toggling the soft action button:

```
To toggle LED states, double tap the RE7 action button within 1 second. Doing so will toggle the LED as:

LED on.
LED off.
```

Would it be possible to expose the turn on and off of the LED in Home Assistant as a light switch? This could be used for automations and night lights ....thanks.

![image](https://user-images.githubusercontent.com/79593761/187035096-98de0def-3021-4064-baf4-80e5e7aeb030.png)

**Comment by kpine** [comment]
Activation of the indicator is not fully supported in HA. All you get are the sensors that are shown in the screen shot. The only place to request new features in the community forums. https://community.home-assistant.io/c/feature-requests/13. You're looking for support to be added for the Indicator Command Class. As there is a variety of functionality provided (the 3 sensor reflect some of the different options), some thought would be required as to how to properly implement it.

In the meantime, you can use low level commands to activate the indicator. In HA you can use the [CC API for Indicator CC](https://zwave-js.github.io/node-zwave-js/#/api/CCs/Indicator?id=indicator-cc-methods) via the [zwave_js.invoke_cc_api](https://www.home-assistant.io/integrations/zwave_js/#service-zwave_jsinvoke_cc_api) service call, or the [zwave_js.set_value service](https://www.home-assistant.io/integrations/zwave_js/#service-zwave_jsset_value) call in HA. The values related to Indicator would be listed in the Device Diagnostic download (click the `...` next to CONFIGURER DEVICE). Here's an example, which I haven't tested because the service call is broken in 2022.8. Maybe automations work.

```yaml
service: zwave_js.invoke_cc_api
data:
  command_class: "135"
  method_name: identify
target:
  entity_id: light.family_room_ceiling
```

You'd have to look in the node-zwave-js code to determine how to use the other CC APIs. I believe the value API maps to the [z-wave spec](https://github.com/zwave-js/specs/blob/master/Indicator%20Command%20Class%2C%20list%20of%20assigned%20indicators%20and%20Property%20IDs%205.9.0.xlsx). 

zwavejs2mqtt allows setting the values in the control panel, or using the CC API via driver code.

**Comment by Anto79-ops** [comment]
Hey @kpine, thanks for your detailed response!!

I can definitely place this on the future request in the forums, i'll do that next.

I have zwavejs2mqtt, makes this a bit easier, BUT figuring out which parameter it is, is another task :) But once I do, I can try using it as a service call using 'zwave_js.invoke_cc_api'

**Comment by Anto79-ops** [comment]
oh, here's the diagnostic file.

[zwave_js-5ec3e01351396fb32e419be34214b2dc-Range Extender 7-899e447661ebb6db420bc2311144aa22.json.txt](https://github.com/zwave-js/node-zwave-js/files/9439849/zwave_js-5ec3e01351396fb32e419be34214b2dc-Range.Extender.7-899e447661ebb6db420bc2311144aa22.json.txt)

**Reply by AlCalzone** [reply]
From that file it looks like the Indicator CC only exposes the **Node Identify** function, which makes the LED blink. You can try asking Aeotec, but I think controlling the LED like you want does not work.

**Reply by Anto79-ops** [reply]
thanks, @AlCalzone I can definitely reach out to Aeotec about this. I won't be too bent out of shape if it doesn't work...lol it would be nice if the repeater had two purposes. 

By the way, I'm still trying to get the information about [the water meter](https://github.com/zwave-js/node-zwave-js/discussions/4902).
